homeassistant/components/esphome/entry_data.py

```diff
@@ -202,13 +243,25 @@ def _unsubscribe() -> None:
         self.assist_pipeline_update_callbacks.append(update_callback)
         return _unsubscribe
 
-    @callback
-    def async_remove_entity(
-        self, hass: HomeAssistant, component_key: str, key: int
-    ) -> None:
+    async def async_remove_entities(self, static_infos: Iterable[EntityInfo]) -> None:
         """Schedule the removal of an entity."""
-        signal = f"esphome_{self.entry_id}_remove_{component_key}_{key}"
-        async_dispatcher_send(hass, signal)
+        coros: list[Coroutine[Any, Any, None]] = []
+        for static_info in static_infos:
+            coro_key = (type(static_info), static_info.key)
+            if coros_for_key := self.entity_info_key_remove_coros.get(coro_key):
+                coros.extend([coros_for_key() for coros_for_key in coros_for_key])
```

Comment:
Should we just call this callbacks?
```suggestion
            callback_key = (type(static_info), static_info.key)
            if callbacks := self.entity_info_key_remove_coros.get(coro_key):
                coros.extend([callback() for callback in callbacks])
```